tools/npm_package_sass_library.bzl

```diff
@@ -0,0 +1,50 @@
+load("@build_bazel_rules_nodejs//:providers.bzl", "NpmPackageInfo")
+load("@io_bazel_rules_sass//:defs.bzl", "SassInfo")
+
+"""
+  Filters a list of files to only return Sass files (with the `.scss` extension).
+"""
+
+def _filter_sass_files(files):
+    return [f for f in files if f.short_path.endswith(".scss")]
+
+"""
+  Rule that extracts Sass sources and its transitive dependencies from a NPM
+  package. The extracted source files are provided with the `SassInfo` provider
+  so that they can be consumed directly as dependencies of other Sass libraries
+  or Sass binaries.
+
+  This rule is helpful when build targets rely on Sass files provided by an external
+  NPM package. In those cases, one wouldn't want to list out all individual source
+  files of the NPM package, but rather glob all needed Sass files from the NPM package.
+"""
+
+def _npm_package_sass_library(ctx):
```

Comment:
Interestingly this could be a useful rule in general for `rules_sass`. We can move that over in a follow-up if we think it makes sense. cc. @jelbourn